data/questions_gen_postgres.csv

```diff
@@ -35,7 +35,7 @@ How many students were predicted to graduate in the last 10 years?,"SELECT count
 How long has it been in days since the last admitted student?,SELECT CURRENT_DATE - max(admit_term) AS duration_since_last_admitted_student FROM student;,advising,date_functions,
 Subtract 2 weeks from the most recent predicted graduation date and give the month.,"SELECT extract(MONTH FROM date_trunc('month', predicted_graduation_semester) - interval '2 weeks') AS MONTH FROM student ORDER BY predicted_graduation_semester DESC LIMIT 1;",advising,date_functions,
 What is the total number of students who found the instructor to be hilarious per course id?,"SELECT course_tags_count.course_id, SUM(course_tags_count.hilarious) AS total_hilarious FROM course_tags_count GROUP BY course_tags_count.course_id;",advising,group_by,
-What is the average clarity score for each instructor who taught a course?,"SELECT {instructor.name, instructor.instructor_id}, avg(course.clarity_score) AS average_clarity_score FROM course JOIN instructor ON course.course_id = instructor.instructor_id GROUP BY {} ORDER BY average_clarity_score DESC NULLS LAST;",advising,group_by,
+What is the average clarity score for each instructor who taught a course?,"SELECT {i.name, i.instructor_id}, AVG(c.clarity_score) FROM course c JOIN course_offering co ON c.course_id = co.course_id JOIN offering_instructor oi ON co.offering_id = oi.offering_id JOIN instructor i ON oi.instructor_id = i.instructor_id GROUP BY {};",advising,group_by,

```

Comment:
we now join from course <> course_offering <> offering_instructor instead of directly from course <> instructor